src/org/labkey/test/WebTestHelper.java

```diff
@@ -359,20 +359,32 @@ public static String buildRelativeUrl(String controller, @Nullable String contai
 
     public static Map<String, String> parseUrlQuery(URL url)
     {
-        String query = url.getQuery();
-        if (query.startsWith("?"))
-            query = query.substring(1);
-        String[] queryArgs = query.split("&");
+        if (url != null)
+            return parseUrlQueryString(url.getQuery());
 
-        Map<String, String> parsedQuery = new HashMap<>(queryArgs.length);
+        return Collections.emptyMap();
```

Comment:
A `null` URL and a URL with no query would return the same thing. I'd distinguish them by returning `null` here.
```suggestion
        return null;
```

Reply:
Actually. Now that I look at the usage, that technique would just cause NPEs in the test. Why do we need to protect against NPE here? A `null` URL seems like it would indicate that something went very wrong.